content/webapp/utils/iiif/v3/index.ts

```diff
@@ -523,6 +528,61 @@ export function groupRanges(
   ).groupedArray;
 }
 
+export const isCanvas = (rangeItem: RangeItems): rangeItem is Canvas => {
+  return typeof rangeItem === 'object' && rangeItem.type === 'Canvas';
+};
+
+export const isRange = (rangeItem: RangeItems): rangeItem is Range => {
+  return typeof rangeItem === 'object' && rangeItem.type === 'Range';
+};
+
+export const isTransformedCanvas = (
+  rangeItem: TransformedRange | TransformedCanvas
+): rangeItem is TransformedCanvas => {
+  return typeof rangeItem === 'object' && rangeItem.type === 'Canvas';
+};
+
+export const isTransformedRange = (
+  rangeItem: TransformedRange | TransformedCanvas
+): rangeItem is TransformedRange => {
+  return typeof rangeItem === 'object' && rangeItem.type === 'Range';
+};
+
+// We want to display the structure from the iiif-manifest
+// But we want to include links, which requires data from the Canvas in the items array
+// So we augment the canvas contained in the structure
+// with the necessary data from the matching canvas item
+
+// TODO write test for this function
```

Comment:
Worth linking to github issues for TODOs so we know the action is accounted for?

Reply:
I'll delete the TODO and capture it in an issue